core/dbt/adapters/factory.py

```diff
@@ -163,10 +164,22 @@ def get_adapter_type_names(self, name: Optional[str]) -> List[str]:
 
 def register_adapter(config: AdapterRequiredConfig) -> None:
     FACTORY.register_adapter(config)
-
-
-def get_adapter(config: AdapterRequiredConfig):
-    return FACTORY.lookup_adapter(config.credentials.type)
+    # Also register the python_adapter if it is specified
+    if config.python_adapter_credentials is not None:
+        FACTORY.register_adapter(config, config.python_adapter_credentials.type)
+
+
+def get_adapter(config: AdapterRequiredConfig, language: str = 'sql'):
+    if language == 'sql':
+        return FACTORY.lookup_adapter(config.credentials.type)
+    elif language == 'python':
+        # TODO: should we add `python_adapter_credentials` to AdapterRequiredConfig?
+        if config.python_adapter_credentials is not None:
```

Comment:
NIT: maybe move this as the first condition (e.g. `if language == 'python' and has_specialized_python_adapter`) so we can we avoid duplicating `FACTORY.lookup_adapter(config.credentials.type)` (since the code path for sql/python would be the same now)